codex-rs/core/src/shell.rs

```diff
@@ -131,6 +134,120 @@ fn strip_bash_lc(command: &Vec<String>) -> Option<String> {
     }
 }
 
+pub fn prettify_command_for_display(command: &[String]) -> Vec<String> {
+    if let Some((script_idx, script)) = extract_shell_script(command)
+        && let Some((_, rest)) = split_cd_prefix_from_script(script)
+    {
+        let mut display = command.to_vec();
+        display[script_idx] = rest;
+        return display;
+    }
+
+    if let Some((_, rest_command)) = split_cd_prefix_from_tokens(command) {
+        return rest_command;
+    }
+
+    command.to_vec()
+}
+
+fn extract_shell_script(command: &[String]) -> Option<(usize, &str)> {
+    match command {
+        [shell, flag, script]
+            if looks_like_shell_wrapper(shell) && (flag == "-lc" || flag == "-c") =>
+        {
+            Some((2, script.as_str()))
+        }
+        _ => None,
+    }
+}
+
+fn looks_like_shell_wrapper(shell: &str) -> bool {
+    let name = Path::new(shell)
+        .file_name()
+        .and_then(|s| s.to_str())
+        .unwrap_or(shell);
+
+    matches!(name, "bash" | "sh" | "zsh")
+}
+
+fn split_cd_prefix_from_script(script: &str) -> Option<(String, String)> {
+    let tree = try_parse_bash(script)?;
+    let mut commands = try_parse_word_only_commands_sequence(&tree, script)?;
+    if commands.is_empty() {
+        return None;
+    }
+
+    commands.reverse();
```

Comment:
As raised on the previous version of this PR, I believe there are other appropriate functions to use that eliminate the need for `reverse()`.